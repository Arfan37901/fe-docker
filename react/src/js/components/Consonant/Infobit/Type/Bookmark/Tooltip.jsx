import React from 'react';
import { string } from 'prop-types';

const TooltipType = { text: string };
const defaultProps = { text: '' };

/**
 * Tooltip (shown in 3:2 Card Footer -- primarily used with bookmark infobit)
 * Used on hover to indicate to users that they do or not do an actino
 *
 * @component
 * @example
 * const props= {
    text: String,
 * }
 * return (
 *   <Tooltip {...props}/>
 * )
 */
const Tooltip = ({ text }) => (
    <span
        className="consonant-tooltip">
        {text}
    </span>
);

Tooltip.propTypes = TooltipType;
Tooltip.defaultProps = defaultProps;

export default Tooltip;
