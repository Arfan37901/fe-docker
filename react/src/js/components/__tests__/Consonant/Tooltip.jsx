import React from 'react';
import renderer from 'react-test-renderer';

import Tooltip from '../../Consonant/Tooltip';

describe('Consonant/Tooltip', () => {
    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const tree = renderer
                .create(<Tooltip text="Tooltip text" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
