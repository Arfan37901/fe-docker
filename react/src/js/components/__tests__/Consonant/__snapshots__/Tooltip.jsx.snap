// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Tooltip Check snapshots should renders correctly 1`] = `
<span
  className="consonant-tooltip"
>
  Tooltip text
</span>
`;
